anx/provider/loadbalancer/address/prefix.go

```diff
@@ -36,6 +41,20 @@ func newPrefix(ctx context.Context, client anxprefix.API, identifier string) (*p
 		ret.family = v1.IPv4Protocol
 	}
 
+	if autoDiscoveryName != nil {
+		tag := fmt.Sprintf("kubernetes-lb-vip-%s", *autoDiscoveryName)
+		vip, err := ret.discoverVIP(ctx, apiclient, ipamClient, tag)
+		if err != nil {
+			// Fall back for backwards compatibility - remove once there are no autodiscovery clusters without tagged VIPs
+			vip = calculateVIP(ret.prefix)
+			logger := logr.FromContextOrDiscard(ctx)
+			logger.Info("Could not auto discover VIP, falling back to calculated VIP", "prefix-identifier", ret.identifier, "vip", vip.String())
+		}
+		ret.addresses = []net.IP{vip}
+	} else {
+		ret.addresses = []net.IP{calculateVIP(ret.prefix)}
+	}
+
```

Comment:
code style: better write this fallback in away we don't have to call `calculateVIP` in two different cases

```go
	if autoDiscoveryName != nil {
		tag := fmt.Sprintf("kubernetes-lb-vip-%s", *autoDiscoveryName)
		vip, err := ret.discoverVIP(ctx, apiclient, ipamClient, tag)
    if err == nil {
   		ret.addresses = []net.IP{vip}
      return &ret, nil
    }
		
    // Fall back for backwards compatibility - remove once there are no autodiscovery clusters without tagged VIPs
		logger := logr.FromContextOrDiscard(ctx)
		logger.Info("Could not auto discover VIP, falling back to calculated VIP", "prefix-identifier", ret.identifier, "vip", vip.String())
		}
	}

  ret.addresses = []net.IP{calculateVIP(ret.prefix)}
  return &ret, nil
```

Reply:
That was a deliberate choice to have a cleanly revertible commit for the deprecated fallback.

Reply:
ah, I'll keep it then